### ISSUE TITLE
Rails 4.2.3 request specs fails with RoutingError when RAILS_RELATIVE_URL_ROOT is set

### ISSUE THREAD

**Issue by dopin** [rails/rails#20740]
I've upgraded to Rails 4.2.3 and ran `bundle ex rspec`.
~~All~~ Some request specs failed with `ActionController::RoutingError: No route matches...`.
Then, I tried `RAILS_RELATIVE_URL_ROOT= bundle ex rspec`, It worked.

Is this a bug? If so, I hope that this will be fixed.

**Comment by sgrif** [comment]
Can you please provide a simple rails app that demonstrates the issue in an isolated fashion?

**Comment by dopin** [comment]
I'm sorry that I should have done so at first.
I made a sample app for this issue: https://github.com/dopin/rails-issues-20740

In short, it seems that it fails if you use 'get post delete` in `config/routes.rb`.
`namespace` also seems failing. I'll add some examples in the sample app later.

By the way, I have created the app by doing `rails-api new`, so this bug might not be rails's bug. I don't know :confused:
